packages/next/src/server/lib/trace/tracer.ts

```diff
@@ -28,7 +29,8 @@ if (process.env.NEXT_RUNTIME === 'edge') {
   }
 }
 
-const { context, trace, SpanStatusCode, SpanKind } = api
+const { context, propagation, trace, SpanStatusCode, SpanKind, ROOT_CONTEXT } =
```

Comment:
Using ROOT_CONTEXT will break when we roll auto registration. Can you use context.active() instead?

Reply:
To change `ROOT_CONTEXT`, this code would have to change: https://github.com/vercel/next.js/blob/254d6d5d6d4cd74ae539313939ee9e994fa9a8d5/packages/next/src/server/lib/trace/tracer.ts#L231-L234

It's probably the right thing to do overall, but would be a traumatic change here. IMHO we should take that on separately from this PR.

Reply:
@agadzik I thought about this a bit more and I actually think using `context.active()` wouldn't be right here. IMHO, if we allow any infra to create parent spans before getting to Next's handler, that infra should be invoking propagation.

Reply:
@agadzik I come up with a simple compromise. The propagation is only done on the root context now. This way it should be always correct in one general case and we can always add support for non-root contexts as we see need later.